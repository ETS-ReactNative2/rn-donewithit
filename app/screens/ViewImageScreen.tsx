import React from 'react';
import {
  Image,
  View,
  SafeAreaView,
  StyleSheet,
  StatusBar,
  Platform,
} from 'react-native';
import { Colors } from 'react-native/Libraries/NewAppScreen';
import colors from '../config/colors';

function ViewImageScreen(props: any) {
  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.infoBox}>
        <View style={styles.tomatoBlock} />
        <View style={styles.turquoiseBlock} />
      </View>
      <Image style={styles.image} resizeMode="contain" source={require('../assets/chair.jpg')} />
    </SafeAreaView>
  );
}

export default ViewImageScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: Platform.OS === 'android' ? StatusBar.currentHeight : 0,
    backgroundColor: colors.black,
  },
  infoBox: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingLeft: 30,
    paddingRight: 30,
  },
  tomatoBlock: {
    width: 50,
    height: 50,
    backgroundColor: colors.primary
  },
  turquoiseBlock: {
    width: 50,
    height: 50,
    backgroundColor: colors.secondary
  },
  image: {
    width: '100%',
    height: '100%',
  },
})
