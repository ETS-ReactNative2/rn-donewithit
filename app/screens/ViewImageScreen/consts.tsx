const CONSTS = {
  ICON_SIZE: 38,
  PADDING_HORIZONTAL: 30
}

export default CONSTS
