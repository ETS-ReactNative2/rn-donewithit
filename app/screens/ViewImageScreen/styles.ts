import { StyleSheet } from 'react-native'
import COLORS from '../../config/colors'
import CONSTS from './consts'

const styles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.BLACK
  },
  infoBox: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: CONSTS.PADDING_HORIZONTAL
  },
  icon: {
    width: CONSTS.ICON_SIZE,
    height: CONSTS.ICON_SIZE
  },
  image: {
    width: '100%',
    height: '100%'
  }
})

export default styles
