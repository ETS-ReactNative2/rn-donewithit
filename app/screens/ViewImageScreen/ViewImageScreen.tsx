import { Image, View } from 'react-native'
import React from 'react'
import CONSTS from './consts'
import AppSafeAreaView from '../../components/AppSafeAreaView'
import AppIcon from '../../components/AppIcon'
import { CHAIR } from '../../config/images'
import styles from './styles'

export default function ViewImageScreen() {
  return (
    <AppSafeAreaView style={styles.container}>
      <View style={styles.infoBox}>
        <AppIcon
          style={styles.icon}
          name="close"
          size={CONSTS.ICON_SIZE}
          color="white"
        />
        <AppIcon
          style={styles.icon}
          name="trash-can-outline"
          size={CONSTS.ICON_SIZE}
          color="white"
        />
      </View>
      <Image style={styles.image} resizeMode="contain" source={CHAIR} />
    </AppSafeAreaView>
  )
}
