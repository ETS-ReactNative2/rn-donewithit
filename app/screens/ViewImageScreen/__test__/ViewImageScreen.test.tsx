import React from 'react'
import renderer from 'react-test-renderer'
import ViewImageScreen from '../index'

describe('ViewImageScreen', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ViewImageScreen />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
