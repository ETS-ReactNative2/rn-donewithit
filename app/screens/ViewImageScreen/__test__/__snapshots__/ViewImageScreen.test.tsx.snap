// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewImageScreen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#000",
      },
      Object {
        "flex": 1,
        "paddingTop": 20,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 30,
      }
    }
  >
    <Text />
    <Text />
  </View>
  <Image
    resizeMode="contain"
    source={1}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</RCTSafeAreaView>
`;
